app/portainer/services/api/stackService.js

```diff
@@ -233,7 +244,7 @@ function StackServiceFactory($q, Stack, ResourceControlService, FileUploadServic
     SwarmService.swarm()
     .then(function success(data) {
       var swarm = data;
-      return FileUploadService.createSwarmStack(name, swarm.Id, stackFile, env, endpointId);
+      return FileUploadService.createSwarmStack(stackDeployOptions.name, swarm.Id, stackFile, stackDeployOptions.env, stackDeployOptions.endpointId);
```

Comment:
The `stackDeployOptions` parameter do not exists in this function. This will cause an error.